src/components/NewCommentForm.tsx

```diff
@@ -59,14 +57,30 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
 
     setSubmitting(true);
 
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
+    if (selectedPost) {
+      const newComment = {
+        name,
+        email,
+        body,
+        postId: selectedPost?.id,
+      };
+
+      const addNewComment = async () => {
+        setSubmitting(true);
+        try {
+          await dispatch(add(newComment));
+        } catch (error) {
+          throw new Error('Adding new comment went wrong');
+        } finally {
+          setSubmitting(false);
+        }
+      };
+
+      addNewComment();
+    }
 
-    // and the spinner will disappear immediately
-    setSubmitting(false);
     setValues(current => ({ ...current, body: '' }));
-    // We keep the entered name and email
-  };
+  }, [[name, email, body, selectedPost]]);
```

Comment:
Remove the double brackets in the dependency array. Change it to [name, email, body, selectedPost].